pkg/deps/config/suppliers.go

```diff
@@ -469,10 +506,13 @@ func newSupplyTxClientsFn(
 	ctx context.Context,
 	deps depinject.Config,
 	signingKeyName string,
-	gasPrices cosmostypes.DecCoins,
+	gasPrices *cosmostypes.DecCoins,
+	gasAdjustment float64,
+	gasSetting *flags.GasSetting,
+	feeAmount *cosmostypes.DecCoins,
```

Comment:
I'm questioning whether it would be better to receive a variadic `TxClientOption` function instead. I'm imagining a use case where you don't care about gas or fees. I think this would be the simplest way to support more use cases.